itou/archive/utils.py

```diff
@@ -26,3 +27,10 @@ def get_year_month_or_none(date=None):
         return timezone.localdate(date).replace(day=1)
 
     return date.replace(day=1)
+
+
+def count_related_subquery(model, fk_field, outer_ref_field, extra_filters=None):
+    filters = {fk_field: OuterRef(outer_ref_field)}
+    if extra_filters:
+        filters.update(extra_filters)
+    return model.objects.filter(**filters).values(fk_field).annotate(count=Count(outer_ref_field)).values("count")
```

Comment:
Tu pourrais retourner 
`Coalesce(..., 0)` (avec to return précédent à la place des ...) pour ne pas avoir à faire du `or 0` un peu partout après.
Et tu peux du coup ne plus l'envelopper dans un `Subquery`, mais directement faire `JobApplication.objects.annotate(truc=count_related_subquery(...))`